source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,994 @@
+        pxDeviceID = &xProvClientHandle->_internal.xRegisterResponse.registration_state.device_id;
+        ulHostnameLength = ( uint32_t ) az_span_size( *pxHostname );
+        ulDeviceIDLength = ( uint32_t ) az_span_size( *pxDeviceID );
+
+        if( ( *pulHostnameLength < ulHostnameLength ) || ( *pulDeviceIDLength < ulDeviceIDLength ) )
+        {
+            AZLogWarn( ( "AzureIoTProvisioning Memory passed is not enough to store hub info " ) );
+            xReturn = eAzureIoTProvisioningFailed;
+        }
+        else
+        {
+            memcpy( pucHubHostname, az_span_ptr( *pxHostname ), ulHostnameLength );
+            memcpy( pucDeviceID, az_span_ptr( *pxDeviceID ), ulDeviceIDLength );
+            *pulHostnameLength = ulHostnameLength;
+            *pulDeviceIDLength = ulDeviceIDLength;
+            xReturn = eAzureIoTProvisioningSuccess;
+        }
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_GetExtendedCode( AzureIoTProvisioningClientHandle_t xProvClientHandle,
+                                                                               uint32_t * pulExtendedErrorCode )
+{
+    AzureIoTProvisioningClientResult_t xReturn;
+
+    if( ( xProvClientHandle == NULL ) ||
+        ( pulExtendedErrorCode == NULL ) )
+    {
+        AZLogError( ( "AzureIoTProvisioning client extended code get failed : Invalid argument" ) );
+        xReturn = eAzureIoTProvisioningInvalidArgument;
+    }
+    else if( xProvClientHandle->_internal.ulWorkflowState != azureiotprovisioningWF_STATE_COMPLETE )
+    {
+        AZLogError( ( "AzureIoTProvisioning client state is not in complete state" ) );
+        xReturn = eAzureIoTProvisioningFailed;
+    }
+    else
+    {
+        *pulExtendedErrorCode =
+            xProvClientHandle->_internal.xRegisterResponse.registration_state.extended_error_code;
+        xReturn = eAzureIoTProvisioningSuccess;
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_SetRegistrationPayload( AzureIoTProvisioningClientHandle_t xProvClientHandle,
+                                                                                      const uint8_t * pucPayload,
+                                                                                      uint32_t ulPayloadLength )
+{
+    AzureIoTProvisioningClientResult_t xReturn;
+
+    if( ( xProvClientHandle == NULL ) ||
+        ( pucPayload == NULL ) || ( ulPayloadLength == 0 ) )
+    {
+        AZLogError( ( "AzureIoTProvisioning client registration payload failed : Invalid argument" ) );
+        xReturn = eAzureIoTProvisioningInvalidArgument;
+    }
+    else if( xProvClientHandle->_internal.ulWorkflowState != azureiotprovisioningWF_STATE_INIT )
+    {
+        AZLogError( ( "AzureIoTProvisioning client state is not in init" ) );
```

Comment:
```suggestion
        AZLogError( ( "AzureIoTProvisioning client state is not initialized" ) );
```

Reply:
INIT state required for set to work properly

---
In reply to: [603655561](https://github.com/Azure/azure-iot-middleware-freertos/pull/49#discussion_r603655561) [](ancestors = 603655561)